client/my-sites/hosting/main.js

```diff
@@ -191,7 +192,7 @@ class Hosting extends Component {
 							<Column type="main" className="hosting__main-layout-col">
 								<SFTPCard disabled={ isDisabled } />
 								<PhpMyAdminCard disabled={ isDisabled } />
-								{ isStagingSiteEnabled && ! isWpcomStagingSite && (
+								{ isStagingSiteEnabled && ! isWpcomStagingSite && ! isProPlan( sitePlanSlug ) && (
```

Comment:
This should work for the hiding staging UI for the Pro plan. What do you think about making this check more generic, to operate on the feature and not on the plan level? That way, we could fix other uncovered cases in which hosting features should be enabled but staging sites shouldn't.

An example of a similar approach could be the `hasSftpFeature` check that exists in the same file.